frame/election-provider-multi-phase/src/lib.rs

```diff
@@ -1401,13 +1401,16 @@ impl<T: Config> Pallet<T> {
 	/// Extracted for easier weight calculation.
 	fn create_snapshot_external(
 	) -> Result<(Vec<T::AccountId>, Vec<VoterOf<T>>, u32), ElectionError<T>> {
-		let target_limit = T::MaxElectableTargets::get().saturated_into::<usize>();
-		let voter_limit = T::MaxElectingVoters::get().saturated_into::<usize>();
+		let election_bounds = T::ElectionBounds::get();
+		let target_limit =
+			election_bounds.targets.count.unwrap_or(u32::MAX).saturated_into::<usize>();
```

Comment:
This is a good point. I think this approach is OK iff the election bounds counts are set, in which case we could default to `0` too. This comes across as somewhat dangerous to me now. I see three options:
1. Fail election if the bounds are not defined;
2. Default to `0` (which probably result in a bad/failed election); or
3. Define sane defaults as consts (e.g. voters: 1000, targets: 1000) and use them as default.

I'm inclined towards option 3., but this may be a silent issue since the election will probably work as expected but the developers may want different bounds (although they are misconfigure). Any ideas?

Reply:
OTOH, we have the size limit tracker putting an upper bound on unbounded (in terms of `count`). But the size bound may also be `None`, which basically means no size limit (in bytes).

This seems to be a configuration issue, so probably we should fail if no election bounds are defined or not set sane election bounds by default if not explicitly given.

Reply:
IMO, the pallet developer should make sure the correct bounds are configured in the EPM pallet, otherwise we'll need to fail the election here, proceed with the election silently with 0 targets/voters or make sure the runtime has sensible bounds, which depends on the case (although this pallet is for internal use only).